go/pkg/rexec/rexec.go

```diff
@@ -364,7 +397,16 @@ func (ec *Context) ExecuteRemotely() {
 		ec.setOutputMetadata()
 		ec.Result = command.NewResultFromExitCode((int)(ec.resPb.ExitCode))
 		if ec.opt.DownloadOutErr {
-			ec.Result = ec.downloadOutErr()
```

Comment:
Why not just add the new option to the boolean guard on the previous line?

Reply:
Sounds like from the user's perspective, streaming overrides downloading, with a fallback to downloading if the server doesn't provide a stream name.

If that's the case, I'd update godocs for both options to clarify this behaviour. Essentially, the godoc should indicate that the download option is ignored if streaming is on, and the streaming option godoc should indicate that the user should expect a delay due to a download fallback if the server doesn't respond to the streaming request.

Reply:
Done.

Reply:
Done.